src/extension/wps/src/main/java/au/org/emii/wps/gogoduck/parameter/SubsetParameters.java

```diff
@@ -39,59 +37,50 @@ public Range getVerticalRange() {
 
     public static SubsetParameters parse(String subset) {
 
-        int timeCount = 0, latLonCount = 0;
-        Pattern timePattern = Pattern.compile("((1[7-9]|20)\\d\\d)-(0?[1-9]|1[012])-(0?[1-9]|[12][0-9]|3[01])T([0-9]+):([0-5]?[0-9]):([0-5]?[0-9]).([0-9]?[0-9]?[0-9]Z)");
-        Pattern latLonPattern = Pattern.compile("([+-]?\\d+\\.?\\d+)\\s*,\\s*([+-]?\\d+\\.?\\d+)");
+        Double latMin, latMax, lonMin, lonMax;
         Map<String, ParameterRange> subsets = new HashMap<>();
-        Matcher matcher;
-
+        String validationErrorMsg = String.format("Invalid latitude/longitude format for subset: %s Valid latitude/longitude format example: LATITUDE,-33.433849,-32.150743;LONGITUDE,114.15197,115.741219", subset);
 
         // Parse
-        for (String part : subset.split(";")) {
-            String[] subsetParts = part.split(",");
-            subsets.put(subsetParts[0], new ParameterRange(subsetParts[1], subsetParts[2]));
+        try {
+            for (String part : subset.split(";")) {
+                String[] subsetParts = part.split(",");
+                subsets.put(subsetParts[0], new ParameterRange(subsetParts[1], subsetParts[2]));
+            }
+        } catch (ArrayIndexOutOfBoundsException e) {
+            throw new GoGoDuckException(validationErrorMsg);
```

Comment:
not necessarily a lat/lon problem here - could be an issue with time or other non supported parameters